hw2/03-count.js

```diff
@@ -1,3 +1,37 @@
 // Add your code here
 
-input.addEventListener('keydown', handleKeyDown);
+const highlightClass = "bg-warning";
+
+const textContainer = document.getElementById("text-container");
+// Restructure the original text container
+const originalTextSpanList = textContainer.innerText.split(" ").map((word) => {
+  const newSpan = document.createElement("span");
+  newSpan.innerText = word;
+  newSpan.style.transitionDuration = "100ms";
+  return newSpan;
+});
+
+textContainer.innerHTML = "";
+originalTextSpanList.forEach((wordSpan) => {
+  textContainer.appendChild(wordSpan);
+  textContainer.appendChild(document.createTextNode(" "));
```

Comment:
Is the purpose of line 17 just to create a space after wordSpan is appended? If so, repeatedly using document.createTextNode() may not be the most straightforward implementation. Since originalTextSpanList is an array it may be more straightforward to use the join() method and format it to include a space between each word.

Reply:
sorry but can you clarify ❓ 

Maybe I miss something but the reason why I did this was that `wordSpan` is an HTML and joining them would make them into a normal string. We can totally parse that string into HTML again but I think it would be a risk for HTML injection

Reply:
Ok